SonarQube.Client/RoslynExporterAdapter/Serializer.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+
+namespace SonarQube.Client.RoslynExporterAdapter
+{
+    /// <summary>
+    /// Helper class to serialize objects to and from XML
+    /// </summary>
+    public static class Serializer
```

Comment:
There is a `Serializer` class in S4MSB, but it has more methods as it also handles loading and saving to disc.
There is a `ToString<T>` method in the S4MSB version, but it only seems to be called by test code (i.e. not used in production). Also, the S4MSB version doesn't correctly save the data as UTF-8.

The upshot is I haven't included a link to the S4MSB `Serializer` class since the implementations are sufficiently different.